### DISCUSSION TITLE
Files not found after migrating from 0.7.9 to 0.9.2

### DISCUSSION THREAD

**Discussion by mjkonko** [pocketbase/pocketbase#1313] [Q&A] [answered]
Hi,
I have run into an issue with files stored after migrating from 0.7.9 to 0.9.2

The background:
I have moved my app's pb_data folder from one server to another (as we have to migrate to go to production officially). Then I ran the migration script. Everything went well and the app has started after some fumbling with the nginx's reverse proxy. I can confirm that PBClient (in the Flutter app) has connection and can in fact download data as I see everything except for the files.

PocketBase is now running on a Hetzner instance (it was before as well just the domain and the machine changed).

The problem in detail:
Logs show 404 when trying to access the file. I have tried reuploading them and there was no change. The Flutter app, using PBClient and `client.records.getFileUrl(record, firstFilename).toString();` does in fact get these urls and they differ by a single slash.

<img width="1699" alt="Screenshot 2022-12-18 at 14 56 53" src="https://user-images.githubusercontent.com/33286755/208302548-fd833b42-2e00-4a80-a798-043acf3e773b.png">

When trying to access the files via a browser a 404 is also returned. I can provide more info, hopefully I'm not missing anything.

<img width="473" alt="Screenshot 2022-12-18 at 15 01 03" src="https://user-images.githubusercontent.com/33286755/208662892-2a261a2f-12e5-49c2-ad2e-ee54b9802681.png">

**Comment by ganigeorgiev** [comment, marked as answer]
PocketBase v0.8+ has breaking changes and requires some manual migration steps.

Please follow the migration guide in https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#migrate-from-v07x.

You can apply the migration steps with the latest v0.10.2 release binary. You'll have to also update your PocketBase Dart SDK to the latest one (`client.records.getFileUrl` is now accessable at `client.getFileUrl`; more info you can find in the related changelog - https://github.com/pocketbase/dart-sdk/blob/master/CHANGELOG.md#050).

**Reply by mjkonko** [reply]
I am most certainly sure that you have I have missed the steps, specifically: updating the Dart SDK. In that case, my apologies, will do that and confirm later on.

**Reply by ganigeorgiev** [reply]
@mjkonko Please note that in addition to the SDK changes, your `pb_data` also will have to be updated to accommodate the new structure. For this purpose there is a temporary `upgrade` command that you can execute that will update your existing collections and API rules (**make sure to have a backup of the directory before running `upgrade`**).

**Reply by mjkonko** [reply]
I did the database migration, just missed the SDK update 🙂 Thank you for the heads up

**Reply by mjkonko** [reply]
As above, I missed the update Dart SDK step. My bad, thanks for pointing it out.
